std-bits/base/src/main/java/org/enso/base/net/URIWithSecrets.java

```diff
@@ -0,0 +1,126 @@
+package org.enso.base.net;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import org.enso.base.enso_cloud.HideableValue;
+import org.graalvm.collections.Pair;
+
+/**
+ * A structure representing a URI that contains parts which may need to be updated once data from
+ * secrets is resolved.
+ *
+ * <p>The query parameters and user info are stored separately, because they may contain secrets and
+ * will only be resolved to plain values within {@link org.enso.base.enso_cloud.EnsoSecretHelper}.
+ */
+public record URIWithSecrets(
```

Comment:
How is this different to `URISchematic`? Why isn't schematic working with `HideableValue` to beging with?

Reply:
The `URISchematic` is just a helper method that I need because I could no longer use the Apache `URIBuilder`. It is encapsulating parts of the logic of building an URI while overriding some parts of it. I did not want to couple this with the Secrets logic as its completely independent concept.

The `URIWithSecrets` may be converted into `URISchematic` in 2 ways - either resolving the secret ids to actual secret values (done only in `EnsoSecretHelper`) OR resolving secrets to just say `__SECRET__` (the 'public' appearance). Then `URISchematic` takes either of these and builds a proper URI.